lib/facter/firewalld_state.rb

```diff
@@ -0,0 +1,14 @@
+def state
+  Facter::Util::Resolution.exec('firewall-cmd --state 2> /dev/null')
```

Comment:
This could use a which to see if it is even installed. Without that, this would cause an error on any linux system that is missing firewall-cmd.